packages/@react-spectrum/menu/docs/Menu.mdx

```diff
@@ -366,6 +367,86 @@ import {Content} from '@react-spectrum/view';
 </MenuTrigger>
 ```
 
+## Submenus <VersionBadge version="alpha" style={{marginLeft: 4, verticalAlign: 'bottom'}} />
+
+Submenus can be created by wrapping an item and a menu in a `SubmenuTrigger`. The `SubmenuTrigger` accepts exactly two children: the `Item` which triggers opening of the submenu, and the `Menu` itself.
+Each submenu's Menu accepts its own set of menu props, allowing you to customize its user action and selection behavior.
+
+### Static
+
+```tsx example
+import {SubmenuTrigger} from '@react-spectrum/menu';
+
+<MenuTrigger>
+  <ActionButton>Actions</ActionButton>
+  <Menu onAction={(key) => alert(`Root menu ${key} action`)}>
+    <Item key="Copy">Copy</Item>
+    <Item key="Cut">Cut</Item>
+    <Item key="Paste">Paste</Item>
+      <SubmenuTrigger>
+        <Item key="Share">Share</Item>
+        <Menu onAction={(key) => alert(`Share menu ${key} action`)}>
+          <Item key="Copy Link">Copy Link</Item>
+          <SubmenuTrigger>
+            <Item key="Email">Email</Item>
+            <Menu onAction={(key) => alert(`Email menu ${key} action`)}>
+              <Item key="Attachment">Email as Attachment</Item>
+              <Item key="Link">Email as Link</Item>
+            </Menu>
+          </SubmenuTrigger>
+          <Item key="SMS">SMS</Item>
+        </Menu>
+      </SubmenuTrigger>
+    <Item key="Delete">Delete</Item>
+  </Menu>
+</MenuTrigger>
+```
+
+### Dynamic
+
+You can define a recursive function to render the nested menu items dynamically.
+
+```tsx example
+import {SubmenuTrigger} from '@react-spectrum/menu';
+
+let items = [
+  {name: 'Copy'},
+  {name: 'Cut'},
+  {name: 'Paste'},
+  {name: 'Share', children: [
+    {name: 'Copy Link'},
+    {name: 'Email', children: [
+      {name: 'Email as Attachment'},
+      {name: 'Email as Link'},
+    ]},
+    {name: 'SMS'},
+  ]},
+  {name: 'Delete'}
+];
+
+let renderSubmenu = (item) => {
+  if (item.children) {
+    return (
+      <SubmenuTrigger>
+        <Item key={item.name}>{item.name}</Item>
+        <Menu items={item.children}>
+          {(item) => renderSubmenu(item)}
+        </Menu>
+      </SubmenuTrigger>
+    );
+  } else {
+    return <Item key={item.name}>{item.name}</Item>;
+  }
+};
+
+<MenuTrigger>
+  <ActionButton>Actions</ActionButton>
+  <Menu items={items}>
+    {(item) => renderSubmenu(item)}
+  </Menu>
+</MenuTrigger>
```

Comment:
You don't have to use an anonymous function for this, so you could shorten this example like so
```suggestion
<MenuTrigger>
  <ActionButton>Actions</ActionButton>
  <Menu items={items}>
    {function renderSubmenu(item) {
      if (item.children) {
        return (
          <SubmenuTrigger>
            <Item key={item.name}>{item.name}</Item>
            <Menu items={item.children}>
              {(item) => renderSubmenu(item)}
            </Menu>
          </SubmenuTrigger>
        );
      } else {
        return <Item key={item.name}>{item.name}</Item>;
      }
    }}
  </Menu>
</MenuTrigger>
```